types/tx/types.go

```diff
@@ -197,3 +197,8 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 func isServiceMsg(typeURL string) bool {
 	return strings.Count(typeURL, "/") >= 2
 }
+
+// ProtoTxProvider is a type which can provide a proto transaction.
+type ProtoTxProvider interface {
```

Comment:
It feels like we're replacing the IntoAny hack with a ProtoTxProvider hack. If you think it's cleaner, then go for it. I don't have a strong preference.

What I would like, for 0.41, is to remove StdTx, so that the Any on TxResponse can be a tx.Tx, and we can remove this hack completely.

Reply:
`IntoAny` was a hack for extracting and serializing a wrapped value. With new constructors we don't need it anymore. `ProtoTxProvider` is not related to `Any`. It's an independent interface - that's why I think it's better. Check for example here: https://github.com/cosmos/cosmos-sdk/pull/8167/files#diff-61a9cb62b0e38da35032d7bc195c38b2c35c342c7302bf6dde5a483f63a96186R458  - `service_test.go:txBuilderToProtoTx`